covalent/_workflow/lepton.py

```diff
@@ -193,19 +198,86 @@ def c_wrapper(*args, **kwargs) -> Any:
             else:
                 return tuple(return_vals)
 
+        def shell_wrapper(*args, **kwargs) -> Any:
+            """Invoke a shell script."""
+
+            import subprocess
+
+            # Call a bash function in a script using library_name and function_name
+            # or
+            # Invoke a generic bash command using only function_name
+
+            if self.function_name == "":
+                raise ValueError(
+                    "A function name or bash command must be provided for a Lepton.function_name."
+                )
+
+            run_lib = f"source {self.library_name} && " if self.library_name != "" else ""
+
+            output_string = ""
+            if kwargs:
+                if "named_outputs" in kwargs:
+                    named_outputs = kwargs["named_outputs"]
+                    del kwargs["named_outputs"]
+
+                    if not isinstance(named_outputs, list):
+                        raise ValueError("Expected a list for Lepton.named_outputs.")
+
+                    for output in named_outputs:
+                        output_string += f" && echo COVALENT-LEPTON-OUTPUT-{output}: ${output}"
+
+                self.function_name = self.function_name.format(**kwargs)
+
+            mutated_args = ""
+            for arg in args:
+                mutated_args += f'"{arg}" '
+
+            shell_cmd = f"{run_lib} {self.function_name} {mutated_args} {output_string}"
+
+            proc = subprocess.run(
+                ["/bin/bash", "-c", shell_cmd],
+                capture_output=True,
+            )
+
+            if proc.returncode != 0:
+                raise Exception(proc.stderr.decode("utf-8").strip())
+
+            return_vals = []
+            if named_outputs:
```

Comment:
I think this line is creating some issues. I get the following error:

```
 File "/Users/faiyaz/Code/covalent/covalent/_workflow/lepton.py", line 246, in shell_wrapper
    if named_outputs:
UnboundLocalError: local variable 'named_outputs' referenced before assignment
```

Reply:
@FyzHsn I'll resolve this later this afternoon

Reply:
should be fixed now